crates/core/component/dex/src/component/position_manager/base_liquidity_index.rs

```diff
@@ -0,0 +1,167 @@
+use anyhow::Result;
+use cnidarium::StateWrite;
+use penumbra_num::Amount;
+use position::State::*;
+
+use crate::lp::position::{self, Position, State};
+use crate::state_key::engine;
+use crate::DirectedTradingPair;
+use penumbra_proto::{StateReadProto, StateWriteProto};
+
+pub(crate) trait AssetByLiquidityIndex: StateWrite {
+    /// Update the base liquidity index, used by the DEX engine during path search.
+    ///
+    /// # Overview
+    /// Given a directed trading pair `A -> B`, the index tracks the amount of
+    /// liquidity available to convert the quote asset B, into a base asset A.
```

Comment:
@erwanor @zbuc maybe this comment could be clarified / expanded in light of today's discussion in the dex channel on discord, explaining that although the index may seem "backwards", we think it's a heuristic that's both useful and also hard to forge?

Reply:
I think we could do this in #4193

Reply:
I think the entire index description could be workshopped, we should put some examples